frontend/javascripts/oxalis/view/right-border-tabs/connectome_tab/connectome_view.js

```diff
@@ -0,0 +1,784 @@
+    const { skeletonId } = this;
+    if (skeletonId != null) {
+      Store.dispatch(removeConnectomeTracingAction(segmentationLayer.name));
+      getSceneController().removeSkeleton(skeletonId);
+      this.skeletonId = null;
+    }
+  }
+
+  resetSkeleton() {
+    const { segmentationLayer } = this.props;
+    if (segmentationLayer == null) return;
+
+    Store.dispatch(removeConnectomeTracingAction(segmentationLayer.name));
+    Store.dispatch(initializeConnectomeTracingAction(segmentationLayer.name));
+  }
+
+  maybeUpdateSkeleton(prevSegmentationLayer?: ?APISegmentationLayer) {
+    const { segmentationLayer } = this.props;
+    if (
+      prevSegmentationLayer != null &&
+      segmentationLayer != null &&
+      prevSegmentationLayer.name === segmentationLayer.name
+    ) {
+      // Although the segmentation layer object identity changed, it is the same layer. This happens
+      // when mapping information is fetched and added to the layer, for example.
+      return;
+    }
+
+    if (prevSegmentationLayer != null) {
+      this.removeSkeleton(prevSegmentationLayer);
+      // Reset the state if the segmentation layer is switched
+      this.reset(prevSegmentationLayer);
+    }
+
+    this.initializeSkeleton();
+  }
+
+  async fetchConnections() {
+    const { dataset, segmentationLayer, currentConnectomeFile, activeAgglomerateIds } = this.props;
+
+    if (
+      currentConnectomeFile == null ||
+      segmentationLayer == null ||
+      activeAgglomerateIds.length === 0
+    )
+      return;
+
+    const fetchProperties = [
+      dataset.dataStore.url,
+      dataset,
+      getBaseSegmentationName(segmentationLayer),
+      currentConnectomeFile.connectomeFileName,
+    ];
+    const synapsesOfAgglomerates = await getSynapsesOfAgglomerates(
+      ...fetchProperties,
+      activeAgglomerateIds,
+    );
+
+    if (synapsesOfAgglomerates.length !== activeAgglomerateIds.length) {
+      throw new Error(
+        `Requested synapses of ${
+          activeAgglomerateIds.length
+        } agglomerate(s), but got synapses for ${synapsesOfAgglomerates.length} agglomerate(s).`,
+      );
+    }
+
+    // Uniquify synapses to avoid requesting data multiple times
+    const allInSynapseIds = unique(synapsesOfAgglomerates.flatMap(connections => connections.in));
+    const allOutSynapseIds = unique(synapsesOfAgglomerates.flatMap(connections => connections.out));
+    const allSynapseIds = unique([...allInSynapseIds, ...allOutSynapseIds]);
+
+    const [
+      synapseSources,
+      synapseDestinations,
+      synapsePositions,
+      synapseTypesAndNames,
+    ] = await Promise.all([
+      getSynapseSources(...fetchProperties, allInSynapseIds),
+      getSynapseDestinations(...fetchProperties, allOutSynapseIds),
+      getSynapsePositions(...fetchProperties, allSynapseIds),
+      getSynapseTypes(...fetchProperties, allSynapseIds),
+    ]);
+    // TODO: Remove once the backend sends the typeToString mapping from the hdf5 file
```

Comment:
todo

Reply:
This cannot be done yet, since the backend jhdf5 library has a bug when reading array attributes. The library author was contacted, but didn't comment yet. I'll extend the TODO comment.